### PR TITLE
PLATUI-872: Add CSRF protection to all non-embedded forms

### PR DESCRIPTION
**Pull request by matthewmascord**

### CODE REVIEW

**Comment by matthewmascord** [comment]
retest this please

**Review by oscarduignan — APPROVED** [review]
Looks good to me 👍

File: it/test/CSRFIntegrationSpec.scala [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+    "respond with 200 for problem_reports_nonjs if a CSRF token is present" in {
+      val getResponse = get(problemReportsNonjsUrl)
+      getResponse.status should be(OK)
+
+      val postResponse = postWithSessionAndCsrfToken(problemReportsNonjsUrl, problemForm, getResponse)
+      postResponse.status should be(OK)
+    }
+
+    "respond with 403 for problem_reports_nonjs if a CSRF token from a different session is provided" in {
```
Comment (JoPintoPaul) [inline it/test/CSRFIntegrationSpec.scala:138]: This is a really good test to have in!

**Comment by matthewmascord** [comment]
Marking as 'Do not merge' as in order for this to be safe to merge, we need to be sure all services on the platform have sameSite == 'lax' and/or we remove the nocsrf Ajax endpoint. Adding nocsrf to an endpoint means there is no longer any check for a X-Requested-With header to allow Ajax requests but not regular form submissions.

**Pull request closed without merging**